app/src/main/java/me/saket/dank/ui/user/UserSessionRepository.java

```diff
@@ -45,6 +61,39 @@ public Completable logout() {
         .andThen(Completable.fromAction(() -> removeLoggedInUsername()));
   }
 
+  public SortingAndTimePeriod getSortingAndTimePeriodForSub(String subredditName) {
+    if (this.sortHashMap.isEmpty())
+      this.loadSortAndPeriod();
+
+    String sortPeriod = this.sortHashMap.get(subredditName);
+
+    if (sortPeriod == null) {
+      return null;
+    } else {
+      return SortingAndTimePeriod.create(sortPeriod);
+    }
+  }
+
+  public void saveSortingAndTimePeriodForSub(String subredditName, SortingAndTimePeriod sortingTime){
+    try {
+      JSONArray arr = new JSONArray();
+
+      this.sortHashMap.put(subredditName, sortingTime.serialize());
+
+      for(String index : this.sortHashMap.keySet()) {
+        JSONObject json = new JSONObject();
+        json.put("subreddit", index);
+        json.put("sort", this.sortHashMap.get(index));
+        arr.put(json);
+      }
+
+      sortingAndPeriod.set(arr.toString());
+
+    } catch (JSONException exception) {
+      // Do something with exception
+    }
+  }
```

Comment:
This project uses [Moshi](https://github.com/square/moshi) for working with JSON :)